docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1735,206 @@ As a response to blocked request AdGuard returns a short video placeholder.
+* [`$stylesheet`](#stylesheet-modifier),
+* [`$subdocument`](#subdocument-modifier),
+* [`$websocket`](#websocket-modifier),
+* [`$xmlhttprequest`](#xmlhttprequest-modifier);
+
+This also includes rules that implicitly add the modifier `$document`:
+* [`$popup`](#popup-modifier);
+
+or special exceptions that implicitly add `$document,subdocument`:
+* [`$content`](#content-modifier),
+* [`$elemhide`](#elemhide-modifier),
+* [`$extension`](#extension-modifier),
+* [`$jsinject`](#jsinject-modifier),
+* [`$specifichide`](#specifichide-modifier),
+* [`$urlblock`](#urlblock-modifier),
+* [`$genericblock`](#genericblock-modifier),
+* [`$generichide`](#generichide-modifier);
+
+
+Specified content-types add `50 + 50 / number_of_content_types`, for example:
+`||example.com^$image,script` will add `50 + 50 / 2 = 50 + 25 = 75` to the total weight of the rule. The `$popup` also belongs to this category, because it implicitly adds the modifier `$document`. Similarly, specific exceptions add `$document,subdocument`.
+
+#### 3. $domain or $app with allowed domains or applications: {#priority-category-3}
+
+Specified domains through `$domain` and specified applications through `$app` add `100 + 100 / number_domains (or number_applications)`, for example:
+`||example.com^$domain=example.com|example.org|example.net` will add `100 + 100 / 3 = 134.3 = 134` or
+`||example.com^$app=org.example.app1|org.example.app2` will add `100 + 100 / 2 = 151`.
+
+#### 4. Specific exceptions: {#priority-category-4}
+
+[//]: # (Please keep them sorted)
+
+* [`$content`](#content-modifier),
+* [`$elemhide`](#elemhide-modifier),
+* [`$extension`](#extension-modifier),
+* [`$jsinject`](#jsinject-modifier),
+* [`$specifichide`](#specifichide-modifier),
+* [`$urlblock`](#urlblock-modifier),
+* [`$genericblock`](#genericblock-modifier),
+* [`$generichide`](#generichide-modifier);
+
+Each of which adds `10^3` to the priority.
+
+As well as
+- exception with a only one modifier (**without any other modifiers**) - [`$document`](#document-modifier) (because it's an alias for `$elemhide,content,jsinject,urlblock,extension`)
+
+In addition, each of the exceptions implicitly adds the two allowed content-type modifiers `$document,subdocument`.
+
+#### 5. Allowlist rule {#priority-category-5}
+
+Modifier `@@` adds `10^4` to rule priority.
+
+#### 6. $redirect rules {#priority-category-6}
+
+The modifier [`$redirect`](#redirect-modifier) adds `10^5` to rule priority.
+Modifier [`$redirect-rule`](#redirect-rule-modifier) adds `10^5 + 1` to rule priority
+The `$redirect-rule` in the presence of a blocking rule is a higher priority than a normal `$redirect`.
+
+#### 7. important rules {#priority-category-7}
+
+Modifier [`$important`](#important-modifier) adds `10^6` to rule priority.
+
+#### Rules for which there is no priority count {#priority-category-extra}
+
+[//]: # (Please keep them sorted)
+
+Rules that do not override blocking, but do additional post- or pre-processing of requests:
```

Comment:
I guess it's better to give here a reference to [the advanced modifiers](https://github.com/AdguardTeam/KnowledgeBase/blob/master/docs/general/ad-filtering/create-own-filters.md#advanced-capabilities) section since all of them don't participate in the priority counting, except for `$all` (which should be mentioned)

Reply:
@sxgunchenko `$redirect` and `$redirect-rules` participate

Reply:
Hm... OK, we can mention that those 3 are handled specially. Still, most of them are not. :)

Reply:
now `$redirect` also added its own weight

Reply:
Yeah, but my point is that it's easy to forget to update the list here introducing a new advanced modifier. That is not a rare case and it is highly likely that the modifier won't participate in the priority counting.

Reply:
@sxgunchenko oh, okay, thanks for the clarification. it seems more correct from this point of view